addons/website_event_track/controllers/webmanifest.py

```diff
@@ -48,8 +48,12 @@ def service_worker(self):
         """ Returns a ServiceWorker javascript file scoped for website_event
         """
         sw_file = get_module_resource('website_event_track', 'static/src/js/service_worker.js')
-        with open(sw_file, 'rb') as fp:
+        with open(sw_file, 'r') as fp:
             body = fp.read()
+        js_cdn_url = 'undefined'
+        if request.website.cdn_activated:
+            js_cdn_url = '"%s"' % request.website.cdn_url
```

Comment:
Side note: the `cdn_url` is off limits to normal users, so it should be safe, but I'm pretty sure someone will report that users with access to website settings can inject any script in `service_worker.js` by playing with `cdn_url`.

One cheap way to prevent that would be something like this:

```suggestion
        if request.website.cdn_activated and request.website.cdn_url:
            cdn_url = request.website.cdn_url.replace('"','%22').replace('\x5c','%5C')
            js_cdn_url = '"%s"' % cdn_url
```

It won't make much difference in practice because we don't expect CDN urls to include a double-quote or backslash char, but it will spare security concerns to some people. Without these 2 chars I don't think you can break out of the JS string literal, right?

Reply:
Completely make sense.
Will change it accordingly.
Thx @odony :sunglasses: